### PR TITLE
Clean the Application name from none-alphanumeric characters for the event controller apps

### PR DESCRIPTION
**Pull request by marcobebway**
**Description**

When reconciling a Kyma subscription, the Application name segment in the event-type must be cleaned from none-alphanumeric characters for:

- The `eventing-controller` app.
- The `eventing-controller-nats` app

**Related issue(s)**
#10218

### CODE REVIEW

File: components/eventing-controller/reconciler/subscription/reconciler_test.go [hunk 1]
```diff
@@ -122,6 +121,11 @@ var _ = Describe("Subscription Reconciliation Tests", func() {
 				Type:    v1.EventTypeWarning,
 			}
 			getK8sEvents(&subscriptionEvents, givenSubscription.Namespace).Should(reconcilertesting.HaveEvent(subscriptionDeletedEvent))
+
+			By("Sending valid webhook requests only")
+			validRequests, invalidRequests := countRequestsForWebhooks(reconcilertesting.EventType, reconcilertesting.EventTypeNotClean)
+			Expect(validRequests == 0).Should(BeTrue())
+			Expect(invalidRequests == 0).Should(BeTrue())
```
Comment (sayanh) [inline components/eventing-controller/reconciler/subscription/reconciler_test.go:128]: You may consider, `Expect(validRequests).Should(Equal(0))` which is more Gomega-ish IMO

File: components/eventing-controller/pkg/handlers/nats_test.go [hunk 2]
```diff
@@ -49,22 +54,22 @@ func TestConvertMsgToCE(t *testing.T) {
...
 			}
 			if tc.expectedErr != nil {
 				if err == nil {
-					t.Errorf("received nil error, expected: %v got: %v", tc.expectedErr, err)
-					return
+					t.Fatalf("Received nil error, expected: %v got: %v", tc.expectedErr, err)
 				}
 				if tc.expectedErr.Error() != err.Error() {
-					t.Errorf("received wrong error, expected: %v got: %v", tc.expectedErr, err)
+					t.Fatalf("Received wrong error, expected: %v got: %v", tc.expectedErr, err)
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply components/eventing-controller/pkg/handlers/nats_test.go:66]: We are using `t.Run` which runs each test case in a separate goroutine, so failing one test case is not affecting the execution of others.

File: components/eventing-controller/reconciler/subscription/reconciler_test.go (hunk 1 above)
Comment (marcobebway) [reply components/eventing-controller/reconciler/subscription/reconciler_test.go]: Done

File: components/eventing-controller/pkg/application/lister.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/informers"
+)
+
+type Lister struct {
+	lister cache.GenericLister
+}
+
+func NewLister(ctx context.Context, client dynamic.Interface) *Lister {
+	gvr := GroupVersionResource()
+	factory := dynamicinformer.NewFilteredDynamicSharedInformerFactory(client, 10*time.Second, v1.NamespaceAll, nil)
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply components/eventing-controller/pkg/application/lister.go]: Done

**Review by sayanh — DISMISSED** [review]

**Review by berkanercan — DISMISSED** [review]

**Review by sayanh — APPROVED** [review]

**Pull request merged**